### PR TITLE
Header redirect/lobby variable port number

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Change Summary & Additional Notes

commit f68abbeb399b7298106b4b88537db9c0ca0bbad3

    Nginx config to redirect to differing lobby depending on header version
    
    Adds an NGINX config that checks the 'triplea-version' header value,
    and based on that value redirects. To support running multiple servers
    on the same host, we need to parameterize the lobby server to be able
    to inject port number. The end goal is we can have multiple lobbies
    running on the same host, different port numbers, and clients will
    be redirected to the correct lobby based on version number in header.

commit 860ed083909beaf1fb39ffe9903cc9454012523a (header-redirect/wip)

    Nginx Config: remove deprecated ssl property
    
    'sudo nginx -t' was generating this warning:
    
    > nginx: [warn] the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /etc/nginx/sites-enabled/default:21

### CODE REVIEW

File: infrastructure/ansible/roles/nginx/templates/etc_nginx_sites_enabled_default.j2 [hunk 1]
```diff
@@ -34,23 +33,17 @@ server {
...
       proxy_set_header        Host $host;
       proxy_set_header        X-Real-IP $remote_addr;
       proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
       proxy_set_header        X-Forwarded-Proto $scheme;
-
-      # Fix the “It appears that your reverse proxy set up is broken" error.
-      proxy_pass          http://localhost:8080;
       proxy_read_timeout  90;
-
-      proxy_redirect      https://localhost:443 https://localhost:8080;
```
Comment (DanVanAtta) [inline infrastructure/ansible/roles/nginx/templates/etc_nginx_sites_enabled_default.j2:47]: This 'proxy_redirect' line is seemingly unneeded. While testing, this line did not seem to matter.

File: infrastructure/ansible/roles/nginx/templates/etc_nginx_sites_enabled_default.j2 [hunk 2]
```diff
@@ -15,7 +15,6 @@ server {
     ssl_certificate {{ cert }} 
     ssl_certificate_key {{ cert_key }}
 
-    ssl on;
```
Comment (DanVanAtta) [inline infrastructure/ansible/roles/nginx/templates/etc_nginx_sites_enabled_default.j2:18]: This 'ssl on' directive is deprecated. This line `listen 443 ssl http2;` is enough to enable SSL

File: infrastructure/ansible/roles/nginx/defaults/main.yml [hunk 3]
```diff
@@ -9,3 +9,19 @@ dhparams_pem_file: /etc/nginx/dhparam.pem
...
   - 443
   - 80
+
+# 'lobby_proxy_pass' captures common selection logic for redirecting clients based on their
+# TripleA Version.
+#
+# Note: header names automatically have dashes converted to underscores.
+# Headers containing underscores are ignored entirely!
+lobby_proxy_pass: |
+    if ($http_triplea_version = "") {
```
Comment (DanVanAtta) [inline infrastructure/ansible/roles/nginx/defaults/main.yml:19]: 'if' is evil in NGINX apparently: https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/

I tried a solution that used `map` to createa `$lobbby_server` variable - though that did not work at all for the websocket `proxy_pass` configurations.

**Review by RoiEXLab — APPROVED** [review]
Looking good, the progress you made there is very impressive

**Pull request merged**